orchestrator/orchestrator.go

```diff
@@ -0,0 +1,213 @@
+package orchestrator
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/ethereum-optimism/supersim/anvil"
+	op_simulator "github.com/ethereum-optimism/supersim/op-simulator"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type ChainConfig struct {
+	Port    uint64
+	ChainId uint64
+	// The base chain ID when the chain is a rollup.
+	// If set to 0, the chain is considered an L1 chain
+	BaseChainId uint64
+}
+
+type OrchestratorConfig struct {
+	ChainConfigs []ChainConfig
+}
+
+type Orchestrator struct {
+	log log.Logger
+
+	OpSimInstances []*op_simulator.OpSimulator
+	anvilInstances []*anvil.Anvil
+}
+
+//go:embed genesisstates/genesis-l1.json
+var genesisL1JSON []byte
+
+//go:embed genesisstates/genesis-l2.json
+var genesisL2JSON []byte
+
+func NewOrchestrator(log log.Logger, config *OrchestratorConfig) (*Orchestrator, error) {
+	var opSimInstances []*op_simulator.OpSimulator
+	var anvilInstances []*anvil.Anvil
+
+	l1Count := 0
+	for _, config := range config.ChainConfigs {
+		if config.BaseChainId == 0 {
+			l1Count++
+		}
+	}
+
+	if l1Count > 1 {
+		return nil, fmt.Errorf("supersim does not support more than one l1")
+	}
+
+	for _, chainConfig := range config.ChainConfigs {
+		var genesis []byte
+		if chainConfig.BaseChainId == 0 {
+			genesis = genesisL1JSON
+		} else {
+			genesis = genesisL2JSON
+		}
```

Comment:
this can be condensed slightly, something like this

```
genesis := genesisL2JSON
if chainConfig.BaseChainId == 0 {
  genesis = genesisL1JSON
}
```